src/Raven.Server/Documents/ConflictsStorage.cs

```diff
@@ -318,21 +319,32 @@ public bool HasHigherChangeVector(DocumentsOperationContext context, Slice prefi
                 var conflictsTable = context.Transaction.InnerTransaction.OpenTable(ConflictsSchema, ConflictsSlice);
                 conflictsTable.DeleteForwardFrom(ConflictsSchema.Indexes[IdAndChangeVectorSlice], prefixSlice, true, long.MaxValue, conflictDocument =>
                 {
-                    var etag = TableValueToEtag((int)ConflictsTable.Etag, ref conflictDocument.Reader);
-                    _documentsStorage.EnsureLastEtagIsPersisted(context, etag);
-
-                    var conflictChangeVector = TableValueToChangeVector(context, (int)ConflictsTable.ChangeVector, ref conflictDocument.Reader);
-                    changeVectors.Add(conflictChangeVector);
+                    var conflicted = TableValueToConflictDocument(context, ref conflictDocument.Reader);
+                    var collection = _documentsStorage.ExtractCollectionName(context, conflicted.Collection);
+
+                    if (conflicted.Doc != null)
+                        _documentsStorage.RevisionsStorage.Put(
+                            context, conflicted.Id, conflicted.Doc, conflicted.Flags | DocumentFlags.Conflicted, nonPersistentFlags, conflicted.ChangeVector,
+                            conflicted.LastModified.Ticks,
+                            collectionName: collection, configuration: RevisionsStorage.ConflictConfiguration.Default);
+                    else if(conflicted.Flags.HasFlag(DocumentFlags.FromReplication) == false)
+                    {
+                        using (Slice.External(context.Allocator, conflicted.LowerId, out Slice key))
+                        {
+                            _documentsStorage.RevisionsStorage.DeleteRevision(context, key, conflicted.Collection, conflicted.ChangeVector);
+                        }
+                    }
+                    _documentsStorage.EnsureLastEtagIsPersisted(context, conflicted.Etag);
+                    changeVectors.Add(conflicted.ChangeVector);
 
-                    var flags = TableValueToFlags((int)ConflictsTable.Flags, ref conflictDocument.Reader);
-                    if ((flags & DocumentFlags.HasAttachments) != DocumentFlags.HasAttachments)
+                    if (conflicted.Flags.HasFlag(DocumentFlags.HasAttachments) == false)
```

Comment:
This does an allocation that the previous code didn't do
https://stackoverflow.com/questions/11665279/why-enums-hasflag-method-need-boxing